src/clients/BundleDataClient/BundleDataClient.ts

```diff
@@ -1077,30 +1043,34 @@ export class BundleDataClient {
             const relayDataHash = getRelayEventKey(slowFillRequest);
 
             if (v3RelayHashes[relayDataHash]) {
-              if (!v3RelayHashes[relayDataHash].slowFillRequest) {
-                v3RelayHashes[relayDataHash].slowFillRequest = slowFillRequest;
-                if (v3RelayHashes[relayDataHash].fill) {
-                  // Exiting here assumes that slow fill requests must precede fills, so if there was a fill
-                  // following this slow fill request, then we would have already seen it. We don't need to check
-                  // for a fill older than this slow fill request.
-                  return;
-                }
-                assert(
-                  isDefined(v3RelayHashes[relayDataHash].deposits) && v3RelayHashes[relayDataHash].deposits!.length > 0,
-                  "Deposit should exist in relay hash dictionary."
-                );
-                const matchedDeposit = v3RelayHashes[relayDataHash].deposits![0];
-
-                if (
-                  slowFillRequest.blockNumber >= destinationChainBlockRange[0] &&
-                  _canCreateSlowFillLeaf(matchedDeposit) &&
-                  !_depositIsExpired(matchedDeposit)
-                ) {
-                  validatedBundleSlowFills.push(matchedDeposit);
-                }
-              } else {
+              if (v3RelayHashes[relayDataHash].slowFillRequest) {
+                this.logger.debug({
+                  at: "BundleDataClient#loadData",
+                  message: "Duplicate slow fill request detected",
+                  slowFillRequest,
+                });
                 throw new Error("Duplicate slow fill request detected.");
               }
+              v3RelayHashes[relayDataHash].slowFillRequest = slowFillRequest;
+              if (v3RelayHashes[relayDataHash].fill) {
+                // Exiting here assumes that slow fill requests must precede fills, so if there was a fill
+                // following this slow fill request, then we would have already seen it. We don't need to check
+                // for a fill older than this slow fill request.
+                return;
+              }
+              assert(
+                isDefined(v3RelayHashes[relayDataHash].deposits) && v3RelayHashes[relayDataHash].deposits!.length > 0,
```

Comment:
I just merged https://github.com/across-protocol/sdk/pull/867, so you should be able to drop the `!` assertion.